src/Stack/Build/Execute.hs

```diff
@@ -229,10 +229,13 @@ executePlan menv bopts baseConfigOpts locals sourceMap plan = do
             , eeGlobalDB = globalDB
             }
 
-    unless (Map.null $ planInstallExes plan) $ do
+    when (installExesPlanFlag $ planInstallExes plan) $ do
+        -- Also install executables
         snapBin <- (</> bindirSuffix) `liftM` installationRootDeps
         localBin <- (</> bindirSuffix) `liftM` installationRootLocal
-        destDir <- asks $ configLocalBin . getConfig
```

Comment:
Why not just have the flag on the command line cause `configLocalBin` to get overridden with the provided value? That way you don't end up with an unused install path floating around that may accidentally get used.